TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODHoldEmCasinoPlugin.cs

```diff
@@ -0,0 +1,2187 @@
+                            else
+                            {
+                                if (!Lobby.IsEmpty()) 
+                                    EnqueueGotoState(VMEODHoldEmCasinoStates.Betting_Round);
+                                else
+                                    EnqueueGotoState(VMEODHoldEmCasinoStates.Waiting_For_Player);
+                            }
+                        }
+                        else
+                            EnqueueGotoState(VMEODHoldEmCasinoStates.Awaiting_Evaluation);
+                        break;
+                    }
+                case VMEODHoldEmCasinoStates.Waiting_For_Player:
+                    {
+                        ActivePlayerIndex = -1;
+                        CardDeck.Shuffle(2);
+                        GameState = newState;
+                        Controller.SendOBJEvent(new VMEODEvent((short)VMEODHoldEmCasinoEvents.Dealer_Collect_Cards));
+                        break;
+                    }
+                case VMEODHoldEmCasinoStates.Awaiting_Evaluation:
+                    {
+                        Lobby.Broadcast("holdemcasino_timer", BitConverter.GetBytes(0));
+                        IdleTimer = 2;
+                        Tock = 0;
+                        GameState = newState;
+                        break;
+                    }
+                case VMEODHoldEmCasinoStates.Pending_Initial_Transactions:
+                    {
+                        Lobby.Broadcast("holdemcasino_timer", BitConverter.GetBytes(0));
+                        IdleTimer = 2;
+                        Tock = 0;
+                        GameState = newState;
+                        break;
+                    }
+                case VMEODHoldEmCasinoStates.Hold:
+                    {
+                        IdleTimer = 3;
+                        Tock = 0;
+                        GameState = newState;
+                        break;
+                    }
+                default:
+                    {
+                        Tock = 0;
+                        GameState = newState;
+                        break;
+                    }
+            }
+        }
+        /*
+         * [WORST CASE SCENARIO ANTE PAYOUT]
+         * Ante Bet (Pay Table 2) => Community cards would be { Ten, Jack, Queen } suited and { 7 or below any suit, same number any suit }
+         * Example community cards { 10c, Jc, Qc, 7s, 7h }, Dealer Qualifies (pair of fours or better) but loses against all players
+         ***********************************************************************************************************************************
+         * Player# - [Type of Hand] - Payout ratio - (desc. of hand) e.g. example
+         ***********************************************************************************************************************************
+         * Player1 - [Royal Flush] - 25:1 - (has King and Ace of same suit as three communit
```

Comment:
Wouldn't this work? (similar for the examples below, though enumeration means you have to use these functions on main thread)
`Players.All(x => (x == null) || Lobby.GetSlotData(x).BetSubmitted)`